src/backend/distributed/utils/multi_partitioning_utils.c

```diff
@@ -87,8 +87,13 @@ fix_pre_citus10_partitioned_table_constraint_names(PG_FUNCTION_ARGS)
 	}
 
 	List *taskList = CreateFixPartitionConstraintsTaskList(relationId);
-	bool localExecutionSupported = true;
-	ExecuteUtilityTaskList(taskList, localExecutionSupported);
+
+	/* do not do anything if there are no constraints that should be fixed */
+	if (taskList != NIL)
```

Comment:
(I think it's much safer to use list_length to check empty lists)

Reply:
The definition of NIL is left untouched for 17 years now :)

I think it is pretty safe to assume there can only be one valid representation of an empty list

pg_list.h:

```
/*
 * The *only* valid representation of an empty list is NIL; in other
 * words, a non-NIL list is guaranteed to have length >= 1.
 */
#define NIL						((List *) NULL)
```